tests/test_basic.py

```diff
@@ -54,34 +55,84 @@ def test_predictor_normal_handling():
                                     mean=[1, 2, 5],
                                     covariance_matrix=C)
 
-# # Test case: update_predictor_beta
-# #
-# # ad2 = ad.AnalyticsDataframe(10000, 3, ["xx1", "xx2", "xx3"], "yy")
-# # ad2.update_predictor_beta(predictor_name_list = ["xx1", "xx2"],
-# #                           a = [1, 2],
-# #                           b = [5, 6])
-
-# # Test case - update predictor normal
-# #
-# ad2 = ad.AnalyticsDataframe(5, 3, ["xx1", "xx2", "xx3"], "yy")
-# ad2 = ad.AnalyticsDataframe(10000, 3, ["xx1", "xx2", "xx3"], "yy")
-# C = np.array([[1, -0.5, 0.3],
-#               [-0.5, 1, 0.2],
-#               [0.3, 0.2, 1]])
-# ad2.update_predictor_normal(predictor_name_list=["xx1", "xx2", "xx3"],
-#                             mean=[1, 2, 5],
-#                             covariance_matrix=C)
-# print(ad2.response_vector)
-# beta = [5, 1, 2, 3]
-# eps_var = 1
-# pred_list = ["xx1", "xx2", "xx3"]
-# ad2.generate_response_vector_linear(beta, eps_var, pred_list)
-# print(ad2.response_vector)
-
-# ##
-# ## Test case - Update categorical values
-# ##
-# # ad2 = AnalyticsDataframe(100, 3, ["xx1", "xx2", "xx3"], "yy")
-# # print(ad2.predictor_matrix)
-# # ad2.update_predictor_categorical("xx2", ["Red", "Green", "Blue"], [0.2, 0.3, 0.5])
-# # print(ad2.predictor_matrix)
+## Test 'update_predictor_beta'
+def test_update_beta():
+    ad = AnalyticsDataframe(10000, 3, ["xx1", "xx2", "xx3"], "yy")
+    ad.update_predictor_beta(predictor_name_list = ["xx1", "xx2"],
+                            a = [1, 2],
+                            b = [5, 6])
+    pred_matrix = ad.predictor_matrix
+    sample_means = pred_matrix.mean().tolist()[:2]
+    sample_vars = np.var(pred_matrix).tolist()[:2]
+    round_decimals = 2
+    xx1_mean = round((1/(1+5)), round_decimals)
+    xx2_mean = round((2/(2+6)), round_decimals)
+    xx1_var = round((1*5)/(((1+5)**2) * (1+5+1)), round_decimals)
+    xx2_var = round((2*6)/(((2+6)**2) * (2+6+1)), round_decimals)
+    assert np.isnan(pred_matrix['xx3'][0]) # nan is of type <class 'numpy.float64'>, has to use isnan().
+    assert [round(sample_mean, round_decimals) for sample_mean in sample_means] == [xx1_mean, xx2_mean]  # same at 3 decimal points
```

Comment:
Following the use of `np.random.seed()` inside function, here, I think we should set some `tolerance` for tests. Apparently, the variance from `generate_beta` function is still slightly varies from theoretical value, which leads to the occasional failure on this tests.
Or we should find a more precise way to generate beta distribution.

Reply:
I think the beta generative function is good enough. The randomness caused by `np.random.beta` determines that we cannot guarantee the sample mean/variance is the same as its theoretical value, especially when the sample size is small.
I think setting some tolerance for the sample mean/variance is a good idea. We have two ways to realize this:
1. Reducing #decimals.
2. $(1 - tolerance) \times mean_{theoretical} \le mean_{sample} \le (1 + tolerance) \times mean_{theoretical}$

I will implement both.